logger_cb.go

```diff
@@ -0,0 +1,157 @@
+package libbpfgo
+
+/*
+#include <bpf/libbpf.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// This callback definition needs to be in a different file from where it is declared in C
+// Otherwise, multiple definition compilation error will occur
+
+// loggerCallback is called by libbpf_print_fn() which in turn is called by libbpf
+//
+//export loggerCallback
+func loggerCallback(libbpfPrintLevel int, libbpfOutput *C.char) {
+	var (
+		level    int
+		goOutput string
+	)
+
+	goOutput = C.GoString(libbpfOutput)
+	goOutput = strings.TrimSuffix(goOutput, "\n")
+
+	for _, fnFilterOut := range callbacks.LogFilters {
+		if fnFilterOut != nil {
+			if fnFilterOut(libbpfPrintLevel, goOutput) {
+				return
+			}
+		}
+	}
+
+	callbacks.Log(level, goOutput)
+}
+
+const (
+	LibbpfWarnLevel  = int(C.LIBBPF_WARN)
+	LibbpfInfoLevel  = int(C.LIBBPF_INFO)
+	LibbpfDebugLevel = int(C.LIBBPF_DEBUG)
+)
+
+type Callbacks struct {
+	Log        func(level int, msg string, keyValues ...interface{})
+	LogFilters []func(libLevel int, msg string) bool
+}
+
+var callbacks = Callbacks{
```

Comment:
A comment saying you're overwriting callbacks if SetLoggerCbs is called would be nice here. I also agree that you should make default LogFilterOuptut generic just like LogFilterLevel (and let tracee implement the output filtering). Maybe open an issue, merge this PR and solve the issue soon.

Reply:
Yeah, I agree. Tackling this soon.

Reply:
@rafaeldtinoco,done.